web-api/src/storage/postgres.rs

```diff
@@ -665,32 +653,27 @@ impl Database {
         tx: &mut Transaction<'_, Postgres>,
         user_id: &UserId,
         time: DateTime<Utc>,
-        interactions: &HashMap<&DocumentId, (&InteractedDocument, UserInteractionType)>,
+        interactions: impl IntoIterator<IntoIter = impl ExactSizeIterator<Item = &DocumentId>>,
     ) -> Result<(), Error> {
+        let mut interactions = interactions.into_iter().peekable();
         //FIXME micro benchmark and chunking+persist abstraction
         let persist = interactions.len() < 10;
 
-        let mut builder = QueryBuilder::new(
-            "INSERT INTO interaction (doc_id, user_id, time_stamp, user_reaction) ",
-        );
-        let mut iter = interactions.iter().peekable();
-        while iter.peek().is_some() {
+        let mut builder =
+            QueryBuilder::new("INSERT INTO interaction (doc_id, user_id, time_stamp) ");
+        while interactions.peek().is_some() {
             builder
                 .reset()
                 .push_values(
-                    iter.by_ref().take(Database::BIND_LIMIT / 4),
-                    |mut builder, (document_id, (_, interaction))| {
+                    interactions.by_ref().take(Database::BIND_LIMIT / 3),
+                    |mut builder, document_id| {
                         builder
                             .push_bind(document_id)
                             .push_bind(user_id)
-                            .push_bind(time)
-                            .push_bind(*interaction as i16);
+                            .push_bind(time);
                     },
                 )
-                .push(
-                    " ON CONFLICT (doc_id, user_id, time_stamp) DO UPDATE SET
-                    user_reaction = EXCLUDED.user_reaction;",
-                )
+                .push(" ON CONFLICT DO NOTHING;")
```

Comment:
we should probably store the newer timestamp on conflict. but this should be handled in a follow up pr, this pr keeps the logic as before.